src/cmd/setup.rs

```diff
@@ -159,44 +160,45 @@ fn setup_dut(repo: &str, cr50: &LocalServo) -> Result<()> {
     }
 
     eprintln!("Setting up DUT: {}", cr50.serial());
-    setup_dut_ccd_open(cr50)?;
-    let mut servo = get_servo_attached_to_cr50(cr50)?;
+    setup_dut_ccd_open(&cr50)?;
     eprintln!("Using Servo: {}", servo.serial());
-    if ensure_dut_network_connection(&mut servo).is_err() {
-        ensure_dev_gbb_flags(repo, cr50)?;
+    if ensure_dut_network_connection(servo).is_err() {
+        ensure_dev_gbb_flags(repo, &cr50)?;
     }
-    let dut = ensure_dut_network_connection(&mut servo)?;
+    let dut = ensure_dut_network_connection(servo)?;
     eprintln!("DUT is ready!");
     eprintln!("export DUT={}", dut.id());
     eprintln!("export SERVO_SERIAL={}", servo.serial());
     Ok(())
 }
 
 #[derive(FromArgs, PartialEq, Debug)]
-/// Setup a DUT for dev
+/// Setup a DUT for dev using Servo
 #[argh(subcommand, name = "dut")]
 pub struct ArgsDut {
+    /// servo serial
+    #[argh(option)]
+    serial: String,
     /// target cros repo dir. If omitted, current directory will be used.
     #[argh(option)]
     repo: Option<String>,
+    /// do ccd unlock only
+    #[argh(switch)]
+    ccd_unlock: bool,
 }
 fn run_dut(args: &ArgsDut) -> Result<()> {
     let repo = get_repo_dir(&args.repo)?;
     ServoList::update()?;
     let list = ServoList::read()?;
-    let list: Vec<&LocalServo> = list.devices().iter().filter(|e| e.is_cr50()).collect();
-    eprintln!("{} Cr50 found.", list.len());
-    if list.is_empty() {
-        return Err(anyhow!(
-        "No Servos or Cr50 are detected. Please check the servo connection, try another side of USB port, attach servo directly with a host instead of via hub, etc..."));
-    }
-    for cr50 in list {
-        if let Err(e) = setup_dut(&repo, cr50) {
-            eprintln!("{:?}", e);
-            continue;
-        }
+    let servo = list.find_by_serial(&args.serial).context(
+        "No Servos or Cr50 are detected. Please check the servo connection, try another side of USB port, attach servo directly with a host instead of via hub, etc...")?;
+    eprintln!("Using {servo:?}");
+    if args.ccd_unlock {
+        let cr50 = get_cr50_attached_to_servo(servo)?;
+        setup_dut_ccd_open(&cr50)
+    } else {
+    setup_dut(&repo, servo)
```

Comment:
nit: miss-indent

Reply:
true, but it is fixed by the later commit. please review this PR as a whole... (I think GitHub's pull-request system is designed in that way, as you can see the diff above is marked as "Outdated") (but thanks for pointing it out!)

Reply:
I doubt the github PR works such way. I think PR is a series of patches because I can not review a whole change. Anyway, it is just a nitpick.